### PR TITLE
Fixed startInlineLogin()

### PR DESCRIPTION
**Pull request by kqwuo**

### CODE REVIEW

**Comment by milleniu** [comment]
Hello!

First of all, I know this is a really small change (+3, -1), however this doesn't mean the provided description can be skipped. Next time please write what was the issue and how the fix does solve the issue.

Anyway, after peer review with @olivier-spinelli:

A regression was introduced when switching from POST to GET method. As described in the BasicDoc when defining the StartLogin process:

> Client applications can easily transfer information through the process:
> -	In inline mode (returnUrl)
> o	The parameters of the returnUrl are kept.
> -	In default mode (popup)
> o	When using GET verb, any query string parameters that are not scheme nor returnUrl are considered as userData.
> o	When using POST verb, any form data are considered as userData.

After the introduced change the userData is now lost. To fix that issue you could use the same query embedded userData which is already used for the `startPopupLogin`. You can also reduce code duplication by adding a single function responsible to map and join an object property when building queries.

Hence, we make the following proposals:
1. Add a `buildQueryString(o: {})` method to build queries.
2. Add a `buildStartLoginUrl` method to build the target url for the StartLogin (either popup or inline) process.
3. Reimplement `startInlineLogin` and `startPopupLogin` methods using the above-mentioned method to always send userData in the query.

File: Clients/webfrontauth/ts/src/AuthService.ts [hunk 1]
```diff
@@ -495,5 +488,12 @@ export class AuthService<T extends IUserInfo = IUserInfo> {
         return this._subscribers.delete(func);
     }
 
+    public buildQueryString( params: { [index: string]: any }, scheme: string ) {
+        return params ? `?scheme=${scheme}&${Object.keys( params ).map((key) => encodeURIComponent(key) + '=' + encodeURIComponent(params[key] as string)).join('&')}` : '';
+    }
+
+    public buildStartLoginUrl( query: string ) {
```
Comment (milleniu) [inline Clients/webfrontauth/ts/src/AuthService.ts]: This method should be private.
You should also add a test to check whether the provided query is empty (it could be the default parameter value).
This method should also be the one responsible for calling `buildQueryString` and should take the param object as parameter.

File: Clients/webfrontauth/ts/src/AuthService.ts [hunk 2]
```diff
@@ -495,5 +488,12 @@ export class AuthService<T extends IUserInfo = IUserInfo> {
         return this._subscribers.delete(func);
     }
 
+    public buildQueryString( params: { [index: string]: any }, scheme: string ) {
```
Comment (milleniu) [inline Clients/webfrontauth/ts/src/AuthService.ts]: This method should be private.
The expected implementation is the one currently used for the the [`sendRequest`](https://github.com/Invenietis/CK-AspNet-Auth/pull/20/files/38c0e1f844fe269a22472f589baf29c3887fc043..8540c15eecddb15bc6be715c1c935ab93e9f0a5b#R199).
It should be called by both the `buildStartLoginUrl` method and the `sendRequest` method.

This will indeed require a little model change for the startLogin process however the send request model is cleaner and allows valueless queries to be provided.

**Review by milleniu — CHANGES_REQUESTED** [review]
See specific comments.

**Review by milleniu — APPROVED** [review]

**Comment by milleniu** [comment]
Good for merge.
